### PR TITLE
Adding notebook for HuggingFaceTF models

### PR DESCRIPTION
**Pull request by 0syrys**
Added a notebook which describes how GradientAccumulateModel can be used with HuggingFace Transformer Models

Also incudes an example training,  runnig on CPU, which shows equivalent training runs with and without GA

### CODE REVIEW

File: notebooks/GA_for_HuggingFace_TF_models.ipynb [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+    "    tf_set = model.prepare_tf_dataset(\n",
+    "      dataset,\n",
+    "      batch_size=batch,\n",
+    "      shuffle=shuffle,\n",
+    "      tokenizer=tokenizer)\n",
+    "\n",
+    "    return tf_set\n",
+    "\n",
+    "#training function\n",
+    "def main(checkpoint,dropout=0.2,batch=8,accum=1,epoch=10,lr=2e-5,seed=42,GA=True):\n",
```
Comment (andreped) [inline notebooks/GA_for_HuggingFace_TF_models.ipynb:168]: Could you rename this function to `run_experiment` or similar, and add spaces between the function arguments, similarly as done before?

Comment (0syrys) [reply notebooks/GA_for_HuggingFace_TF_models.ipynb]: > Could you remove the *.pkl file, as I assume it is generated by the program and not needed to run this notebook, @0syrys?

Its actually the input data for the experiment


> Could you rename this function to `run_experiment` or similar, and add spaces between the function arguments, similarly as done before?

Done

Comment (andreped) [reply notebooks/GA_for_HuggingFace_TF_models.ipynb]: Great! Good job :)

**Comment by andreped** [comment]
I see that the *.pkl file still remains. See [here](https://github.com/0syrys/GradientAccumulator/tree/GA-HuggingFace/notebooks) in your own fork.

Please, remove this file in a new commit, and then I believe this PR can be merged.

**Comment by 0syrys** [comment]
> Could you remove the *.pkl file, as I assume it is generated by the program and not needed to run this notebook, @0syrys?

The .pkl File is actually the input for the experiment
If you want I can remove it anyway but the example won't run obviously

**Comment by andreped** [comment]
> > Could you remove the *.pkl file, as I assume it is generated by the program and not needed to run this notebook, @0syrys?
> 
> The .pkl File is actually the input for the experiment If you want I can remove it anyway but the example won't run obviously

Oh, OK. I thought it was something that was generated by the notebook. The file does not seem that large. Then I will add it.

Great job!

**Comment by andreped** [comment]
@0syrys Could you set your full name for your GitHub user name, such that it follows the appropriate style in the Zenodo citation? See:
https://zenodo.org/record/7890319#.ZFIhAuxBweY

For compare our profiles to see how they differ to see what to change:
https://github.com/andreped

**Pull request merged**